pkg/gsclient/gsclient.go

```diff
@@ -0,0 +1,181 @@
+package gsclient
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"errors"
+	"net/http"
+	"net/url"
+
+	"github.com/Masterminds/semver/v3"
+	gsclient "github.com/giantswarm/gsclientgen/v2/client"
+	"github.com/giantswarm/gsclientgen/v2/client/auth_tokens"
+	"github.com/giantswarm/gsclientgen/v2/client/clusters"
+	"github.com/giantswarm/gsclientgen/v2/client/key_pairs"
+	"github.com/giantswarm/gsclientgen/v2/models"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/go-openapi/runtime"
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+)
+
+type Config struct {
+	Logger micrologger.Logger
+
+	// Installations configuration
+	Email    string
+	Endpoint string
+	Password string
+}
+
+type Client struct {
+	authWriter runtime.ClientAuthInfoWriter
+	email      string
+	endpoint   string
+	password   string
+
+	client *gsclient.Gsclientgen
+}
+
+var providers = []string{
+	"aws",
+}
+
+func AllProviders() []string {
+	return providers
+}
+
+func IsValidProvider(candidate string) bool {
+	for _, provider := range providers {
+		if candidate == provider {
+			return true
+		}
+	}
+	return false
+}
+
+func IsValidRelease(candidate string) bool {
+	_, err := semver.NewVersion(candidate)
+	return err == nil
+}
+
+func New(config Config) (*Client, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.Email == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Email must not be empty", config)
+	}
+
+	if config.Password == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Password must not be empty", config)
+	}
+
+	if config.Endpoint == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Endpoint must not be empty", config)
+	}
+
+	u, err := url.Parse(config.Endpoint)
+	if err != nil {
+		return nil, microerror.Maskf(invalidConfigError, "API endpoint URL %s could not be parsed", config.Endpoint)
+	}
+
+	tlsConfig := &tls.Config{}
+	transport := httptransport.New(u.Host, "", []string{u.Scheme})
+	transport.Transport = &http.Transport{
+		Proxy:           http.ProxyFromEnvironment,
+		TLSClientConfig: tlsConfig,
+	}
+	gsClient := gsclient.New(transport, strfmt.Default)
+
+	client := Client{
+		endpoint: config.Endpoint,
+		email:    config.Email,
+		password: config.Password,
+		client:   gsClient,
+	}
+
+	return &client, nil
+}
+
+func (c *Client) CreateCluster(ctx context.Context, releaseVersion string, provider string) (string, error) {
```

Comment:
Mostly pulled from https://github.com/giantswarm/gsctl/blob/master/client/client.go Maybe this could be moved into a common library.

Reply:
It would be cool to be able to use it directly from `gsctl`. Might be worth the effort to refactor over there in the long run